### PR TITLE
 warm states for V02

### PR DESCRIPTION
**Pull request by jhrehanoaa**
Looking for insight here. We have several functions that interact with the computation here. I am not sure which parts equate to what in v02 for warm states. Every time I want to add something it's connected elsewhere.

### CODE REVIEW

File: src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py [hunk 1]
```diff
@@ -386,7 +427,7 @@ def main():
             jobs = []
             for twi, (tw, reach_list) in enumerate(reaches_bytw.items(), 1):
                 r = list(chain.from_iterable(reach_list))
-                param_df_sub = param_df.loc[
+                channel_initial_states_df = param_df.loc[
```
Comment (awlostowski-noaa) [inline src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py:430]: this seems to overwrite channel_initial_states_df, which contains our initial condition data?

File: src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py [hunk 2]
```diff
@@ -396,9 +437,9 @@ def main():
                         nts,
                         reach_list,
                         independent_networks[tw],
-                        param_df_sub.index.values,
-                        param_df_sub.columns.values,
-                        param_df_sub.values,
+                        channel_initial_states_df.index.values,
+                        channel_initial_states_df.columns.values,
+                        channel_initial_states_df.values,
```
Comment (awlostowski-noaa) [inline src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py:442]: this should still ref param_df_sub

Comment (awlostowski-noaa) [reply src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py:442]: at line 434, 
`
q0_sub = channel_initial_states_df.loc[r].sort_index()
`

**Review by awlostowski-noaa — COMMENTED** [review]
seems like the initial state data are getting slotted into the wrong spot.

File: src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py (hunk 2 above)
Comment (jameshalgren) [reply src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py:442]: @jhrehanoaa -- rebased to latest upstream/master

**Comment by jameshalgren** [comment]
@jhrehanoaa -- rebased to latest upstream/master

File: src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py [hunk 3]
```diff
@@ -333,6 +338,34 @@ def main():
...
+
+    if wrf_hydro_channel_restart_file:
+
+        q0 = nhd_io.get_stream_restart_from_wrf_hydro(
+            wrf_hydro_channel_restart_file,
+            wrf_hydro_channel_ID_crosswalk_file,
+            wrf_hydro_channel_ID_crosswalk_file_field_name,
+            wrf_hydro_channel_restart_upstream_flow_field_name,
+            wrf_hydro_channel_restart_downstream_flow_field_name,
+            wrf_hydro_channel_restart_depth_flow_field_name,
```
Comment (jameshalgren) [inline src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py:356]: We'll need to add commandline arguments to populate these variables (or something like them...)

**Review by jameshalgren — APPROVED** [review]

**Pull request merged**